### DISCUSSION TITLE
Powering down and powering up

### DISCUSSION THREAD

**Discussion by andrewjsmith76** [PiSupply/PiJuice#849] [Q&A]
Hi!

Love the PiJuice. I purchased as part of the original Kickstarter campaign and played around with it, but haven't had a project for it until now... So I was pleasantly surprised to see it in my Pi-drawer when I needed it!

I've been poking around in the settings, but I can't seem to get the behaviour that I want, which is:
- when power is lost, power the Pi from battery until it reaches x% capacity (something high-ish, say: 80%)
- when battery level drops under x%, power down the pi and cut power
- when external power is restored, boot up the pi

The first step is easy. I could use the watchdog to power up using the 'wakeup on charge' option.
I was hopeful that the "power present" System Event would do the trick, but there isn't a 'wakeup' option there.

Does anybody have an idea?

Thanks in advance!

Best regards,
Andrew

**Comment by andrewjsmith76** [comment]
... forgot to add: I tried the option to wakeup upon charged to x% - but that leaves a gap between the time when power is restored until the pi actually starts booting again - or do I simply set the wakeup charge level to something way lower than the shutdown level?

My fear is that if the battery drains further due to a prolonged power outage, the pi will be stuck in a shutdown-restart-loop...

**Reply by tvoverbeek** [reply]
Since this is a Kickstarter original version, the first thing to do is to upgrade the pijuice-base and pihuice-gui packages and then to upgrade the microcontroller firmware.
Current packages version is 1.8 and current firmware version is 1.6.
The latest software version (1.8) runs on both 32-bit and 64-bit OS. Previous versions only on 32-bit.

- Set Wakeup On charge (System Task panel) to a low level (e.g. 1%) and enable the Restore option. This will store the threshold in non-volatile memory. If not the threshold can be lost under certain circumstances. This will eake up the pi as soon as charging occurs, i.e. when external power returns
- Set Min charge (System Task panel) to 80%
- On System events panel  enable Low Charge and select SYS_FUNC_HALT_POW_OFF. Note the lo charge event only occurs when no external power is present.

I believe this will do what you want.

**Reply by andrewjsmith76** [reply]
Hello Tvoverbeek!

Thank you for getting back to me so guickly! I did update the firmware (today) and did an apt-get upgrade, so I believe I have the current version... I was also playing around with pijuice_cli and now the icon in the desktop's upper right-hand corner that shows what is currently going on with the power is gone... 

I'll try the settings you suggest.

Thanks again!

Best regards,
Andrew

**Comment by andrewjsmith76** [comment]
... The settings work exactly as I need them to. Thanks!
